### ISSUE TITLE
Setting timeout zero does not disable the timeout (its setting timeout to be zero)

### ISSUE THREAD

**Issue by anderslangballe** [RediSearch/RediSearch#1987]
I am running Redisearch 2.0.7 in Docker. When my index grows large, the `FT.AGGREGATE` behaves in ways I do not expect.

For example, running the following command as inspired by the tutorial:
```
FT.AGGREGATE search42 "*"
  APPLY "@startTime - (@startTime % 3600)" AS hour

  GROUPBY 1 @hour
    REDUCE COUNT_DISTINCT 1 @event_id AS num_events

  SORTBY 2 @hour ASC
```

... returns the following at sub-200K documents:
```
 1) (integer) 15
 2) 1) "hour"
    2) "1618653600000"
    3) "num_events"
    4) "29926"
 3) 1) "hour"
    2) "1618653603600"
    3) "num_events"
    4) "33"
 4) 1) "hour"
    2) "1618740000000"
    3) "num_events"
    4) "24975"
...
```

... but at around 200K documents, begins to return the following:
```
1) (integer) 216899
2) (empty array)
(0.56s)
```

`FT.SEARCH` seems to work as expected at this index size. Any ideas what could cause this, or what limits may need to be tweaked? `maxmemory` is set to 0, and the host has around 5 GB free memory when this problem occurs.

**Comment by anderslangballe** [comment]
It seems that I am getting a timeout at this index size. Setting `TIMEOUT` to `0`, which should disable the timeout according to [docs](https://oss.redislabs.com/redisearch/Configuring/#timeout), did not resolve this. A higher timeout seems to solve my problem.

**Comment by MeirShpilraien** [comment]
@anderslangballe you are right, looking at the code I see that setting timeout zero is actually setting it to zero and not disabling it. Thanks for reporting. I will modify the issue title to better explain the issue.

**Comment by ashtul** [comment]
fixed by #2012
